contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -0,0 +1,133 @@
+		ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler,
+	) (
+		out middleware.InitializeOutput, metadata middleware.Metadata, err error,
+	) {
+		// Bind the timestamp to the context so that we can use it when we have enough information to start the trace.
+		ctx = context.WithValue(ctx, spanTimestampKey{}, time.Now())
+		return next.HandleInitialize(ctx, in)
+	}), middleware.Before)
+}
+
+func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
+	return stack.Initialize.Add(middleware.InitializeMiddlewareFunc("StartTraceMiddleware", func(
+		ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler,
+	) (
+		out middleware.InitializeOutput, metadata middleware.Metadata, err error,
+	) {
+		operation := awsmiddleware.GetOperationName(ctx)
+		serviceID := awsmiddleware.GetServiceID(ctx)
+
+		opts := []ddtrace.StartSpanOption{
+			tracer.SpanType(ext.SpanTypeHTTP),
+			tracer.ServiceName(serviceName(mw.cfg, serviceID)),
+			tracer.ResourceName(fmt.Sprintf("%s.%s", serviceID, operation)),
+			tracer.Tag(tagAWSRegion, awsmiddleware.GetRegion(ctx)),
+			tracer.Tag(tagAWSOperation, operation),
+			tracer.Tag(tagAWSService, serviceID),
+			tracer.StartTime(ctx.Value(spanTimestampKey{}).(time.Time)),
+		}
+		if !math.IsNaN(mw.cfg.analyticsRate) {
+			opts = append(opts, tracer.Tag(ext.EventSampleRate, mw.cfg.analyticsRate))
+		}
+		span, spanctx := tracer.StartSpanFromContext(ctx, fmt.Sprintf("%s.request", serviceID), opts...)
+
+		// Handle initialize and continue through the middleware chain.
+		out, metadata, err = next.HandleInitialize(spanctx, in)
+		span.Finish(tracer.WithError(err))
+
+		return out, metadata, err
+	}), middleware.After)
+}
+
+func (mw *traceMiddleware) deserializeTraceMiddleware(stack *middleware.Stack) error {
+	return stack.Deserialize.Add(middleware.DeserializeMiddlewareFunc("DeserializeTraceMiddleware", func(
+		ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler,
+	) (
+		out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+	) {
+		span, _ := tracer.SpanFromContext(ctx)
+
+		// Get values out of the request.
+		if req, ok := in.Request.(*smithyhttp.Request); ok {
+			span.SetTag(ext.HTTPMethod, req.Method)
+			span.SetTag(ext.HTTPURL, req.URL.String())
+			span.SetTag(tagAWSAgent, req.Header.Get("User-Agent"))
+		}
+
+		// Continue through the middleware chain which eventually sends the request.
+		out, metadata, err = next.HandleDeserialize(ctx, in)
+
+		// Get values out of the response.
+		if res, ok := out.RawResponse.(*smithyhttp.Response); ok {
+			span.SetTag(ext.HTTPCode, res.StatusCode)
+		}
+
+		// Extract the request id.
+		if requestID, ok := awsmiddleware.GetRequestIDMetadata(metadata); ok {
+			span.SetTag(tagAWSRequestID, requestID)
+		}
+
+		return out, metadata, err
+	}), middleware.Before)
```

Comment:
Changed this to `Before`. Actually missed test coverage on GetRequestIDMetadata and there was a bug. The AWS middleware for setting the request ID was happening after our middleware.